File: TownOfSalemCalculator/src/townofsalemcalculator/Conditions/Concrete/NightMessages/Transported.java
package townofsalemcalculator.Conditions.Concrete.NightMessages;

import java.util.Arrays;
import java.util.List;
import static townofsalemcalculator.Role.Transporter;
import townofsalemcalculator.Conditions.RoleGroup.SingleRoleGroup;

/**
 * The Concrete Condition that you were transported
 * @author Multifacio
 * @version 1.1
 * @since 2017-1-28
 */
public class Transported extends NightMessage {
    public Transported() {
        super(new SingleRoleGroup(Transporter));
    }
    
    @Override
    public String getCondition() {
        return "You were transported to another location";
    }

    @Override
    public List<String> keyWords() {
        List<String> keyWords = Arrays.asList(new String[]{"Transported", "Transed", "Another", "Location", "You"});
        keyWords.addAll(Transporter.getKeyWords());
        return keyWords;
    }
}

File: src/Conditions/Concrete/Effects/Transported.py
from src.Concepts.Role import Role
from src.Conditions.Concrete.ExistenceCondition import ExistenceCondition
from typing import FrozenSet

class Transported(ExistenceCondition):
    """ The Transported condition is used when during the night you were transported (by a Transported), i.e. you
    received the message: 'You were transported to another location' """

    def __init__(self, amnesiacRemembered: FrozenSet[Role] = None):
        super().__init__(frozenset({Role.TRANSPORTER}), amnesiacRemembered)